src/wp-includes/option.php

```diff
@@ -1774,33 +1675,11 @@ function update_network_option( $network_id, $option, $value ) {
 		return add_network_option( $network_id, $option, $value );
 	}
 
-	$notoptions_key = "$network_id:notoptions";
-	$notoptions     = wp_cache_get( $notoptions_key, 'site-options' );
-
-	if ( is_array( $notoptions ) && isset( $notoptions[ $option ] ) ) {
-		unset( $notoptions[ $option ] );
-		wp_cache_set( $notoptions_key, $notoptions, 'site-options' );
-	}
-
 	if ( ! is_multisite() ) {
 		$result = update_option( $option, $value, 'no' );
 	} else {
-		$value = sanitize_option( $option, $value );
-
-		$serialized_value = maybe_serialize( $value );
-		$result           = $wpdb->update(
-			$wpdb->sitemeta,
-			array( 'meta_value' => $serialized_value ),
-			array(
-				'site_id'  => $network_id,
-				'meta_key' => $option,
-			)
-		);
-
-		if ( $result ) {
-			$cache_key = "$network_id:$option";
-			wp_cache_set( $cache_key, $value, 'site-options' );
-		}
+		$value  = sanitize_option( $option, $value );
+		$result = update_metadata( 'site', $network_id, $option, $value );
```

Comment:
```suggestion
		$result = update_metadata( 'site', $network_id, wp_slash( $option ), wp_slash( $value ) );
```

ref

https://github.com/WordPress/wordpress-develop/blob/deafd1193c244acfae80bb4507de4d8b532fa9e5/src/wp-includes/meta.php#L203-L207

Reply:
🥴

Reply:
Pushed a fix https://github.com/WordPress/wordpress-develop/pull/2478/commits/faf732ada8e0f3092ffabb3e7d013b778dfade39